### PR TITLE
Replace printStackTrace() with a proper logger

### PR DESCRIPTION
**Pull request by Yeregorix**
I originally had an issue where my plugin failed to construct but I had no traces in the log files so I wanted to replace the printStackTrace() with a proper logger. I took the opportunity to do this in several places.

### CODE REVIEW

**Review by gabizou — APPROVED** [review]
There's a fair few places we call `e.PrintStackTrace()`, mind fixing those up in one fell swoop?

**Comment by Yeregorix** [comment]
There are still other `printStackTrace()` indeed.
Most of them are in the test plugins / unit tests, I thought it wasn't really important.
Some others are in asm class visitors so wasn't sure if it would cause classloading issues or not.

**Comment by Yeregorix** [comment]
I have now replaced all `printStackTrace` except in test sourcesets, and except in `SpongeWorldManager` because it will be fixed by PR #3702.

File: src/main/java/org/spongepowered/common/block/entity/SpongeBlockEntityArchetypeBuilder.java [hunk 1]
```diff
@@ -99,7 +102,7 @@ public BlockEntityArchetype.Builder from(final BlockEntityArchetype value) {
     public BlockEntityArchetype.Builder state(final BlockState state) {
         final net.minecraft.world.level.block.state.BlockState blockState = (net.minecraft.world.level.block.state.BlockState) state;
         if (!((BlockStateBridge) (blockState)).bridge$hasTileEntity()) {
-            new IllegalArgumentException("BlockState: "+ state + " does not provide TileEntities!").printStackTrace();
+            SpongeBlockEntityArchetypeBuilder.LOGGER.warn("BlockState " + state + " does not provide TileEntities!", new IllegalArgumentException());
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/block/entity/SpongeBlockEntityArchetypeBuilder.java]: TileEntities -> BlockEntities

Comment (zml2008) [reply src/main/java/org/spongepowered/common/block/entity/SpongeBlockEntityArchetypeBuilder.java]: you'll also want to replace string concat with string interpolation

```suggestion
            SpongeBlockEntityArchetypeBuilder.LOGGER.warn("BlockState {} does not provide TileEntities!", state, new IllegalArgumentException());
```

**Comment by Zidane** [comment]
@Yeregorix @zml2008 

I am aware at the amount of effort of such a change but zml you are 100% right, the "global logger" is not helpful, not to know where whatever is coming from. I think it is time to likely axe it so `SpongeCommon.logger()` goes away and each class that logs creates it's own logger. Thoughts? Beyond being a massive change, I see no issue. This doesn't change giving each plugin a logger. If a plugin wants to do class loggers or not that is up to them. This is just a stylistic choice for us to keep doing this for them.

**Comment by Yeregorix** [comment]
From a developer point of view, I prefer each class having its own logger.
From a server owner pov, it's easier to identify which mod is involded when you see `[Sponge]` at the beginning of the logged lines.

1. `[Sponge] Something went wrong.` (easy to identify that Sponge is involved)
vs
2. `[SomeClass] Something went wrong.` (most people have no idea what `SomeClass` is or to which mod it belongs)

**Comment by Zidane** [comment]
@Yeregorix I agree. However classes such as DataUpdaterDelegate fall right into this issue in this PR, no? I'd have to look at our typical log files but someone wouldn't know based on the logging lines.

**Comment by Yeregorix** [comment]
Yes, I'm just opposing the pros and cons of each convention.

**Comment by Yeregorix** [comment]
For the moment, for each class I used the common logger if I saw that it was already used in the class, otherwise I used a specific logger.

**Review by zml2008 — APPROVED** [review]

**Pull request merged**